docs/components/index.md

```diff
@@ -564,6 +564,133 @@ Otherwise, the type parameter must be explicitly specified using an attribute th
 </ListViewTemplate>
 ```
 
+## Cascading values and parameters
+
+In some scenarios, it's inconvenient to flow data from an ancestor component to a descendent component using [component parameters](#component-parameters), especially when there are several component layers. Cascading values and parameters solve this problem by providing a convenient way for an ancestor component to provide a value to all of its descendent components. Cascading values and parameters also provide an approach for components to coordinate.
+
+In the following example from the sample app, the `ThemeInfo` class specifies the theme information to flow down the component hierarchy so that all of the buttons within a given part of the app share the same style.
+
+*UIThemeClasses/ThemeInfo.cs*:
+
+```csharp
+public class ThemeInfo
+{
+    public string ButtonClass { get; set; }
+}
+```
+
+An ancestor component can provide a cascading value using the `CascadingValue` component. The `CascadingValue` component wraps a subtree of the component hierarchy and specifies a single value to all components within that subtree.
+
+For example, the sample app specifies theme information (`ThemeInfo`) in one of the app's layouts as a cascading parameter for all components that make up the layout body of the `@Body` property. `ButtonClass` is assigned a value of `btn-success` in the layout component. Any descendent component can consume this property through the `ThemeInfo` cascading object.
+
+*Shared/CascadingValuesParametersLayout.cshtml*:
+
+```cshtml
+@inherits BlazorLayoutComponent
+@using BlazorSample.UIThemeClasses
+
+<div class="container-fluid">
+    <div class="row">
+        <div class="col-sm-3">
+            <NavMenu />
+        </div>
+        <div class="col-sm-9">
+            <CascadingValue Value="@theme">
+                <div class="content px-4">
+                    @Body
+                </div>
+            </CascadingValue>
+        </div>
+    </div>
+</div>
+
+@functions {
+    ThemeInfo theme = new ThemeInfo { ButtonClass = "btn-success" };
+}
+```
+
+To make use of cascading values, components declare cascading parameters using the `[CascadingParameter]` attribute. Cascading values are bound to cascading parameters by type.
```

Comment:
They can also be bound based on a string name value, e.g.,

    [CascadingParameter(Name = "mysupertheme")]

This is relevant if you have multiple cascading values of the same type and need to differentiate them.